src/logging.cc

```diff
@@ -833,6 +833,91 @@ void LogDestination::DeleteLogDestinations() {
   sinks_ = NULL;
 }
 
+namespace {
+
+vector<string> SplitString(const string& s, const char delimiter) {
+  std::stringstream ss(s);
+  string t;
+  vector<string> tokens;
+
+  while (std::getline(ss, t, delimiter)) {
+    if (t.length() > 0) {
+      tokens.push_back(t);
+    }
+  }
+  return tokens;
+}
+
+bool IsGlogLog(const string& log_name) {
+  static const int kGlogFilenameTokenCount = 6;
+  vector<string> log_name_tokens = SplitString(log_name, '.');
```

Comment:
If `log_name` contains a `.`, then this will be broken

Reply:
Resolved by passing `ent->d_name` into IsGlogLog() directly so that we don't have to perform filename stripping ourselves, which also solves this problem.